Cubical/Categories/Adjoint.agda

```diff
@@ -0,0 +1,296 @@
+{-# OPTIONS --cubical --no-import-sorts --safe #-}
+
+module Cubical.Categories.Adjoint where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Data.Sigma
+open import Cubical.Categories.Category
+open import Cubical.Categories.Functor
+open import Cubical.Categories.NaturalTransformation
+open import Cubical.Foundations.Isomorphism
+
+open Functor
+
+open Iso
+open Precategory
+
+{-
+==============================================
+                  Overview
+==============================================
+
+This module contains two definitions for adjoint
+functors, and functions witnessing their
+logical (and maybe eventually actual?)
+equivalence.
+-}
+
+private
+  variable
+    ℓC ℓC' ℓD ℓD' : Level
+
+-- I tried using this... but Agda can never infer the category,
+-- even when it's pretty clear from context.....
+-- _⋆_ : ∀ {ℓ ℓ'} {C : Precategory ℓ ℓ'} {x y z} (f : C [ x , y ]) (g : C [ y , z ]) → C [ x , z ]
+-- _⋆_ {C = C} = seq C
+
+{-
+==============================================
+             Adjoint definitions
+==============================================
+
+We provide two alternative definitions for
+adjoint functors: the unit-counit
+definition, followed by the natural bijection
+definition.
+-}
+
+-- Adjoint def 1: unit-counit
+record _⊣_ {C : Precategory ℓC ℓC'} {D : Precategory ℓD ℓD'} (F : Functor C D) (G : Functor D C)
+                 : Type (ℓ-max (ℓ-max ℓC ℓC') (ℓ-max ℓD ℓD')) where
+  field
+    -- unit
+    η : 𝟙⟨ C ⟩ ⇒ (funcComp G F)
+    -- counit
+    ε : (funcComp F G) ⇒ 𝟙⟨ D ⟩
+    -- triangle identities
+    Δ₁ : PathP (λ i → NatTrans (F-lUnit {F = F} i) (F-rUnit {F = F} i))
+       (seqTransP {p = F-assoc} (F ∘ʳ η) (ε ∘ˡ F))
+       (1[ F ])
+    Δ₂ : PathP (λ i → NatTrans (F-rUnit {F = G} i) (F-lUnit {F = G} i))
+       (seqTransP {p = sym F-assoc} (η ∘ˡ G) (G ∘ʳ ε))
+       (1[ G ])
+
+-- Adjoint def 2: natural bijection
+record _⊣²_ {C : Precategory ℓC ℓC'} {D : Precategory ℓD ℓD'} (F : Functor C D) (G : Functor D C) : Type (ℓ-max (ℓ-max ℓC ℓC') (ℓ-max ℓD ℓD')) where
+  field
+    adjIso : ∀ {c d} → Iso (D [ F ⟅ c ⟆ , d ]) (C [ c , G ⟅ d ⟆ ])
```

Comment:
What is the notation around c and d?

Reply:
It is a functor's action on an object. I picked it somewhat arbitrarily, is there a better/standard notation for this?

Reply:
I don't really like this notation, but I don't think there's anything standard for it. Maybe just calling the field `ob` or `fob` or something so that you can write `G .ob d` or `G .fob d`?

Reply:
Hmm. I felt like when working with things it was nice to have a mixfix notation for functor actions to make it "feel" more like working with category theory on paper. But yeah, if the notation makes it confusing I can change it in the next PR (after we sort out this build issue).

Reply:
Ok, sounds good!